ginga/canvas/types/basic.py

```diff
@@ -158,7 +159,7 @@ def contains_arr(self, x_arr, y_arr):
         # See: http://alienryderflex.com/polygon/
         xa, ya = x_arr, y_arr
 
-        result = numpy.empty(ya.shape, dtype=numpy.bool)
+        result = numpy.empty((ya.size, xa.size), dtype=numpy.bool)
```

Comment:
@ejeschke , I don't know why, the `xa` and `ya` are actually `xi` and `yi` in `BaseImage.get_shape_view()`:
```python
x1, y1, x2, y2 = map(int, shape_obj.get_llur())

if avoid_oob:
    # avoid out of bounds indexes
    wd, ht = self.get_size()
    x1, x2 = max(0, x1), min(x2, wd-1)
    y1, y2 = max(0, y1), min(y2, ht-1)

# calculate pixel containment mask in bbox
yi = numpy.mgrid[y1:y2+1].reshape(-1, 1)
xi = numpy.mgrid[x1:x2+1].reshape(1, -1)
contains = shape_obj.contains_arr(xi, yi)  #  <-------- CRASHES HERE
```